hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsAsyncQueue.kt

```diff
@@ -0,0 +1,15 @@
+package uk.gov.justice.hmpps.sqs
+
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+import software.amazon.awssdk.services.sqs.model.GetQueueUrlRequest
+
+class HmppsAsyncQueue(
+  val id: String,
+  val sqsAsyncClient: SqsAsyncClient,
+  val queueName: String,
+  val sqsAsyncDlqClient: SqsAsyncClient? = null,
+  val dlqName: String? = null
+) {
+  val queueUrl: String by lazy { sqsAsyncClient.getQueueUrl(GetQueueUrlRequest.builder().queueName(queueName).build()).get().queueUrl() }
+  val dlqUrl: String? by lazy { sqsAsyncDlqClient?.getQueueUrl(GetQueueUrlRequest.builder().queueName(dlqName).build())?.get()?.queueUrl() }
+}
```

Comment:
One large change in AWS SDK V2 is that the asynchronous SQS/SNS clients no longer extend the synchronous client. This feature was heavily used previously as the async client could do the job of the sync client. Now we have to handle the async clients separately.